cloudinit/config/cc_grub_dpkg.py

```diff
@@ -62,22 +65,37 @@ def handle(name, cfg, _cloud, log, _args):
     idevs_empty = util.get_cfg_option_str(
         mycfg, "grub-pc/install_devices_empty", None)
 
-    if ((os.path.exists("/dev/sda1") and not os.path.exists("/dev/sda")) or
-       (os.path.exists("/dev/xvda1") and not os.path.exists("/dev/xvda"))):
+    # get the root disk where the /boot directory resides.
+    # the ESP will reside on the same root disk as /boot, but not device.
+    try:
+        disk = util.subp(['grub-probe', '-t', 'disk', '/boot'],
+                         status_cb=False, capture=True)[0].strip()
```

Comment:
The default value of `status_cb` is `None`, so I don't know that we need to pass `False` in here.

Reply:
I actually thought `status_cb` did something completely different, but upon reading the implementation of uitl.subp() I was wrong. Latest commit drops `status_cb` completely.